internal/pkg/apiutil/attribute.go

```diff
@@ -147,6 +151,116 @@ func NewClusterListAttributeFromNative(a *bgp.PathAttributeClusterList) *api.Clu
 	}
 }
 
+func NewPrefixSIDAttributeFromNative(a *bgp.PathAttributePrefixSID) *api.PrefixSID {
+	psid := &api.PrefixSID{}
+	psid.Tlvs = MarshalSRv6TLVs(a.TLVs)
+
+	return psid
+}
+
+func MarshalSRv6TLVs(tlvs []bgp.PrefixSIDTLVInterface) []*any.Any {
+	mtlvs := make([]*any.Any, 0)
+	for _, tlv := range tlvs {
+		var r proto.Message
+		switch t := tlv.(type) {
+		case *bgp.SRv6L3ServiceAttribute:
+			o := &api.SRv6L3ServiceTLV{}
+			o.SubTlvs = MarshalSRv6SubTLVs(t.SubTLVs)
+			r = o
+		default:
+			log.WithFields(log.Fields{
+				"Topic": "protobuf",
+				"SRv6":  t,
+			}).Warn("invalid prefix sid tlv type to marshal")
+			return nil
+		}
+		a, _ := ptypes.MarshalAny(r)
+		mtlvs = append(mtlvs, a)
+	}
+
+	return mtlvs
+}
+
+func MarshalSRv6SubTLVs(tlvs []bgp.PrefixSIDTLVInterface) map[uint32]*api.SRv6TLV {
+	mtlvs := make(map[uint32]*api.SRv6TLV)
+	var key uint32
+	for _, tlv := range tlvs {
+		var r proto.Message
+		switch t := tlv.(type) {
+		case *bgp.SRv6InformationSubTLV:
+			o := &api.SRv6InformationSubTLV{
+				EndpointBehavior: uint32(t.EndpointBehavior),
+				// TODO Once flags are used in RFC, add processing.
+				Flags: &api.SRv6SIDFlags{},
+			}
+			o.Sid = make([]byte, len(t.SID))
+			copy(o.Sid, t.SID)
+			o.SubSubTlvs = MarshalSRv6SubSubTLVs(t.SubSubTLVs)
+			// SRv6 Information Sub TLV is type 1 Sub TLV
+			key = 1
+			r = o
+		default:
+			log.WithFields(log.Fields{
+				"Topic": "protobuf",
+				"SRv6":  t,
+			}).Warn("invalid prefix sid sub tlv type to marshal")
+			return nil
+		}
+		a, _ := ptypes.MarshalAny(r)
+		_, ok := mtlvs[key]
+		if !ok {
+			mtlvs[key] = &api.SRv6TLV{
+				Tlv: make([]*any.Any, 0),
+			}
+		}
+		tlvs := mtlvs[key]
+		tlvs.Tlv = append(tlvs.Tlv, a)
+		mtlvs[key] = tlvs
+	}
+
+	return mtlvs
+}
+
+func MarshalSRv6SubSubTLVs(tlvs []bgp.PrefixSIDTLVInterface) map[uint32]*api.SRv6TLV {
+	mtlvs := make(map[uint32]*api.SRv6TLV)
+	var key uint32
+	for _, tlv := range tlvs {
+		var r proto.Message
+		switch t := tlv.(type) {
+		case *bgp.SRv6SIDStructureSubSubTLV:
+			o := &api.SRv6StructureSubSubTLV{
+				LocalBlockLength:    uint32(t.LocalBlockLength),
+				LocalNodeLength:     uint32(t.LocatorNodeLength),
+				FunctionLength:      uint32(t.FunctionLength),
+				ArgumentLength:      uint32(t.ArgumentLength),
+				TranspositionLength: uint32(t.TranspositionLength),
+				TranspositionOffset: uint32(t.TranspositionOffset),
+			}
+			// SRv6 SID Structure Sub Sub TLV is type 1 Sub Sub TLV
+			key = 1
+			r = o
+		default:
+			log.WithFields(log.Fields{
+				"Topic": "protobuf",
+				"SRv6":  t,
+			}).Warn("invalid prefix sid sub sub tlv type to marshal")
+			return nil
+		}
+		a, _ := ptypes.MarshalAny(r)
+		_, ok := mtlvs[key]
+		if !ok {
+			mtlvs[key] = &api.SRv6TLV{
+				Tlv: make([]*any.Any, 0),
+			}
+		}
+		tlvs := mtlvs[key]
```

Comment:
this snippet can be rewritten to have one map lookup less, like:
```
tlvs, ok := mtlvs[key]
if !ok {
  tlvs = &api.SRv6TLV{
    Tlv: make([]*any.Any, 0),
  }
  mtlvs[key] = tlvs
}
```

so we get rid of an extra lookup in line 256 after the `if` block.

Reply:
done